### PR TITLE
Throw on box change

### PR DESCRIPTION
**Pull request by fweik**
Preliminary throw an error if the box size is changed with active constraints,
this does not work properly with all shapes and can have some non-intuiitive
side effects. Also removed CONSTRAINT feature and moved constraints out
of the short range loop.

### CODE REVIEW

**Comment by jonaslandsgesell** [comment]
Some test cases might still contain a check for the constraint feature. E.g. constraint_shape_based.py

**Comment by fweik** [comment]
Ready for review.

File: src/core/constraints/Constraints.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      for (auto const &c : *this) {
+        c->add_energy(p, pos, energy);
+      }
+    }
+  }
+
+  void on_boxl_change() const {
+    if (not this->empty()) {
+      throw std::runtime_error("The box size can not be changed because there "
+                               "are active constraints. " +
```
Comment (KaiSzuttor) [inline src/core/constraints/Constraints.hpp]: Can you please change the order of the strings, we get a nicer error message then.

**Review by KaiSzuttor — APPROVED** [review]
LGTM besides comment

**Pull request merged**